src/vs/workbench/contrib/terminal/browser/links/terminalLinkManager.ts

```diff
@@ -97,6 +97,20 @@ export class TerminalLinkManager extends DisposableStore {
 		this._registerStandardLinkProviders();
 	}
 
+	async getLinks(type: TerminalLinkProviderType, y: number): Promise<ILink[] | undefined> {
+		let provider: ILinkProvider | undefined = undefined;
+		if (type === TerminalLinkProviderType.Word) {
+			provider = this._standardLinkProviders.find(p => p instanceof TerminalWordLinkProvider);
+		} else if (type === TerminalLinkProviderType.Validated) {
+			provider = this._standardLinkProviders.find(p => p instanceof TerminalValidatedLocalLinkProvider);
+		} else {
+			provider = this._standardLinkProviders.find(p => p instanceof TerminalProtocolLinkProvider);
+		}
+		if (provider === undefined) {
+			throw new Error(`no link provider of type ${type}`);
+		}
+		return (await new Promise<ILink[] | undefined>(r => provider?.provideLinks(y, r)))!;
```

Comment:
`provider` definitely exists here but TS doesn't know since inside the callback `provider` could have been reset. We should refactor this such that the return type of `getLinks` is `Promise<ILink[]>`, I don't think undefined is a possible result as it will throw when there is no provider.